website/client/store/getters/tasks.js

```diff
@@ -109,3 +116,51 @@ export function getTaskClasses (store) {
     }
   };
 }
+
+// Returns all list for given task type
+export function getUnfilteredTaskList ({state}) {
+  return (type) => state.tasks.data[`${type}s`];
+}
+
+// Returns filtered, sorted, ordered, tag filtered, and search filtered task list
+// @TODO: sort task list based on used preferences
+export function getFilteredTaskList ({state, getters}) {
+  return ({
+    type,
+    filterType = '',
+  }) => {
+    // get requested tasks
+    // check if task list has been passed as override props
+    // assumption: type will always be passed as param
+    let requestedTasks = getters['tasks:getUnfilteredTaskList'](type);
+
+    let userPreferences = state.user.data.preferences || {};
```

Comment:
these defaults aren't necessary, the api always return these fields

Reply:
updated as suggested